File: NodeRoot.java
import java.util.ArrayList;

public class NodeRoot extends Node{
    public ArrayList<Node> listNode;

    public NodeRoot(int number){
        super(number);
        Node aux = this.getThis();
        listNode = new ArrayList<Node>();
        listNode.add(aux);
    }

    public Node[] getNodes(){
        
        int listSize = this.listNode.size();
        Node[] nodeArray = new Node[listSize];
        for(int i =0; i < listSize;i++){
            nodeArray[i] = listNode.get(i);
        }
        return nodeArray;
    }
}

File: PHP/NodeRoot.php
<?php 

require_once("Node.php");

class NodeRoot extends Node{
    public $listNode;

    function __construct(int $number){
        //super(number);
        parent::__construct($number);
        $aux = parent::getThis();
        $this->listNode = [];
        array_push($this->listNode,$aux);
    }

    function addList(Node $node){
        array_push($this->listNode,$node);
    }

    function getNodes(){
        
        $listSize = count($this->listNode);
        $nodeArray = [];
        for($i = 0; $i < $listSize; $i++){
            $nodeArray[$i] = $this->listNode[$i];
        }
        return $nodeArray;
    }

    function addNode(int $number){
        $newNode = parent::addNode($number);
        $this->addList($newNode);
    }
    

}